### ISSUE TITLE
[BUG] Types issues with Transactions in typescript

### ISSUE THREAD

**Issue by madhusudhand** [dynamoose/dynamoose#924]
### Summary:

The suggested example when used with TypeScript it gives couple issues because of types mismatch.

```js
await dynamoose.transaction([
    User.transaction.update({"id": "user1"}, {"$ADD": {"balance": -100}}),
    Charge.transaction.create({"userid": "user1", "product": "product1", "amount": 100, "status": "successful"}),
    Product.transaction.update({"id": "product1"}, {"$ADD": {"inventory": -1}}),
    Credit.transaction.delete({"id": "credit1"}),
    {
        "Delete": {
            "Key": {
                "id": {
                    "S": "helloworld"
                }
            },
            "TableName": "MyOtherTable"
        }
    }
]);
```

### Current output and behavior (including stack trace):

1. TypeScript complains about 3 required params for dynamoose.transaction even though others have defaults set in the library.

2. Model.transaction doesn't exists for TypeScript as it isn't directly defined on Model object. So, User.transaction is an error for typescript.

However adding ts-ignore it works as the actual implementation exists in JS objects.

### Expected output and behavior:

Fix is needed providing proper types for Model.transaction.

### Environment:

Operating System: Mac
Operating System Version: 10.14
Node.js version (`node -v`): 12.13
NPM version: (`npm -v`): 6.12
Dynamoose version: latest aws

### Other:
- [x] I have read through the Dynamoose documentation before posting this issue
- [x] I have searched through the GitHub issues (including closed issues) and pull requests to ensure this issue has not already been raised before
- [x] I have searched the internet and Stack Overflow to ensure this issue hasn't been raised or answered before
- [x] I have tested the code provided and am confident it doesn't work as intended
- [x] I have filled out all fields above
- [x] I am running the latest version of Dynamoose

**Comment by fishcharlie** [comment]
Sounds like just TypeScript typings issues. No underlying problems here. Pull requests would be amazing!

**Comment by fishcharlie** [comment]
I think this is solved as of #991
